lib/api/v3/work_packages/form/work_package_attribute_links_representer.rb

```diff
@@ -41,58 +41,60 @@ class WorkPackageAttributeLinksRepresenter < Roar::Decorator
 
           self.as_strategy = ::API::Utilities::CamelCasingStrategy.new
 
-          property :status,
-                   exec_context: :decorator,
-                   getter: -> (*) {
-                     { href: api_v3_paths.status(represented.status_id) }
-                   },
-                   setter: -> (value, *) {
-                     resource = parse_resource(:status, :statuses, value['href'])
-
-                     represented.status_id = resource[:id] if resource
-                   }
-
-          property :assignee,
-                   exec_context: :decorator,
-                   getter: -> (*) {
-                     id = represented.assigned_to_id
-
-                     { href: (api_v3_paths.user(id) if id) }
-                   },
-                   setter: -> (value, *) {
-                     user_id = parse_user_resource(:assignee, value['href'])
-
-                     represented.assigned_to_id = user_id
-                   }
-
-          property :responsible,
-                   exec_context: :decorator,
-                   getter: -> (*) {
-                     id = represented.responsible_id
-
-                     { href: (api_v3_paths.user(id) if id) }
-                   },
-                   setter: -> (value, *) {
-                     user_id = parse_user_resource(:responsible, value['href'])
-
-                     represented.responsible_id = user_id
-                   }
-
-          property :version,
-                   exec_context: :decorator,
-                   getter: -> (*) {
-                     id = represented.fixed_version_id
-
-                     { href: (api_v3_paths.version(id) if id) }
-                   },
-                   setter: -> (value, *) {
-                     resource = parse_resource(:version, :versions, value['href'])
-
-                     represented.fixed_version_id = resource[:id] if resource
-                   }
+          def self.linked_property(property_name: nil,
```

Comment:
@myabc they are not optional at the moment. But I believe that when you have more than 2 (3 in edge cases) parameters in your signature, naming those parameters makes calls to the method easier to read. 

Additionally, I think that they might become optional. There is a lot of similarity between the 4 parameter values (e.g. statuses, status, status_id) thus one could apply some intelligent guessing within the method later. Having the signature use named parameters will then allow us to make some of the parameters optional without having to refactor all the calls. I simply didn't get to it yesterday.